test/test_integration_systemd.rb

```diff
@@ -0,0 +1,64 @@
+require_relative "helper"
+require_relative "helpers/integration"
+
+require 'sd_notify'
+
+class TestIntegrationSystemd < TestIntegration
+  def setup
+    skip "Skipped on Windows because it does not support Systemd" if windows?
+    skip UNIX_SKT_MSG unless UNIX_SKT_EXIST
+
+    ::Dir::Tmpname.create("puma_socket") do |sockaddr|
+      @sockaddr = sockaddr
+      @socket = Socket.new(:UNIX, :DGRAM, 0)
+      socket_ai = Addrinfo.unix(sockaddr)
+      @socket.bind(socket_ai)
+      ENV["NOTIFY_SOCKET"] = sockaddr
+    end
+
+    ENV["SD_NOTIFY"] = "1"
+  end
+
+  def teardown
+    return if skipped?
+    @socket.close if @socket
+    File.unlink(@sockaddr) if @sockaddr
+    @socket = nil
+    @sockaddr = nil
+    ENV["SD_NOTIFY"] = nil
+    ENV["NOTIFY_SOCKET"] = nil
+  end
+
+  def socket_message
+    @socket.recvfrom(10)[0]
+  end
+
+  def test_notify_protocol
```

Comment:
I don't see a purpose to this test, because you're just testing the behavior of the `SdNotify` gem, which, presumably, has it's own tests.